app/routes.py

```diff
@@ -1,2 +1,284 @@
+
+    # is_complete = task.completed_at
+    # if task.completed_at != None:
+    #     is_complete == True
+    # print(is_complete)
+
+    if request.method == "GET":
+        select_task = {
+            "task": {
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at
+            }
+        }
+        return jsonify(select_task), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f"Task {task.id} \"{task.title}\" successfully deleted"})
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        completed_at = form_data["completed_at"]
+
+        db.session.commit()
+
+        return {
+            "task": {
+                "id": task.id,
+                "title": "Updated Task Title",
+                "description": "Updated Test Description",
+                "is_complete": True if task.completed_at else False
+            }
+        }, 200
+
+
+def post_slack(message_slack):
+
+    TOKEN_SLACK = os.environ.get(
+        "SLACK_BOT_TOKEN")
+    slack_path = "https://slack.com/api/chat.postMessage"
+    query_params = {
+        'channel': 'task-notifications',
+        'text': message_slack
+    }
+    headers = {'Authorization': f"Bearer {TOKEN_SLACK}"}
+    requests.post(slack_path, params=query_params, headers=headers)
+
+
+@task_list_api_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response(f"{task_id} doesnt exist", 404)
+    task.completed_at = datetime.utcnow()
+
+    db.session.commit()
+
+    if request.method == "PATCH":
+        message_slack = f"Someone just completed the task: {task.title}"
+        post_slack(message_slack)
+        return {
+            "task": {
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": True if task.completed_at else False
+            }
+        }, 200
+
+
+@task_list_api_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response(f"{task_id} doesnt exist", 404)
+
+    task.completed_at = None
+    db.session.commit()
+
+    if request.method == "PATCH":
+        return {
+            "task": {
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete":  True if task.completed_at else False
+            }
+        }, 200
+
+
+@goals_bp.route("", methods=["GET", "POST"])
+def handle_goals():
```

Comment:
Consider also breaking up this GET and POST for goals into two helper functions `get_goals` and `create_goal`